jar/walker.go

```diff
@@ -133,6 +135,10 @@ func (w *walker) skipDir(path string, d fs.DirEntry) bool {
 }
 
 func (w *walker) visit(p string, d fs.DirEntry) error {
+	if w.Backup {
+		BackupFiles(w.filepath(p))
```

Comment:
This needs to return its error. It should also probably only be after the directory and extension checks